lib/build/OCaml.om

```diff
@@ -614,8 +614,8 @@ public.OCamlOpt() =
                 $(OCamlOpt) -o $@ -c $<
         elseif $(BYTE_ENABLED)
             %.cmx %.cmi %$(EXT_OBJ) %.cmo: %.ml :scanner: scan-ocaml-%.ml
-                $(OCamlC) -o $@ -c $<
-                $(OCamlOpt) -o $(replacesuffixes .cmx, $(EXT_OBJ), $@) -c $<
+                $(OCamlC) -o $(replacesuffixes .cmx, .cmo, $@) -c $<
```

Comment:
I think this `-o` is ignored anyway... at least the compiler does not complain when you want to output a `.cmx` from `ocamlc` (which isn't possible, actually). However, as noted in the description of the PR, we run into problems because the commands generated by this rule are not identical to the commands when a `.o` file needs to be generated (below).

@cspiel What was the motivation for adding `-o`?

Reply:
Requesting a cmx file from OCamlC is blunder and we must
replace the suffixes (cmx -> cmo) in the call to the byte compiler.

I just cannot retrace why you have dropped the OCamlOpt call as
this is the branch where both the byte compiler and the native
compiler are called.   Do I get yet another thing wrong, or are
we missing
```
        $(OCamlOpt) -o $@ -c $<
```

Also compare with the `BYTE_ENABLED` branch of the
rule `%$(EXT_OBJ): %.ml`.

Reply:
oh, I did not drop the ocamlopt call, see (new) line 618

Reply:
@gerdstolpmann you are right.  I tripped over the "compressed"
diff format.  LGTM.